src/ngraph/runtime/gpu/gpu_external_function.cpp

```diff
@@ -254,6 +254,31 @@ runtime::gpu::GPU_ExternalFunction::~GPU_ExternalFunction()
 {
 }
 
+void runtime::gpu::GPU_ExternalFunction::assemble_and_reserve()
+{
+    GPUAllocator allocator = m_shared_context->m_primitive_emitter->get_memory_allocator();
```

Comment:
Shouldn't from line 259 through 273 be a pass? Seems like you are working hard to make it not a pass. The pass manager state is really intended as an communications channel between passes. You could make this a function pass if you wanted. Since this code runs immediately after the pass manager just make this a pass.

Reply:
#1246 is a WIP towards converging on a way in which all runtime allocations (tensor pool and kernel-based memory) can be exported to the graph. In offline discussions we decided to iterate on it a bit more and to use #1231 until we converge on how to best achieve this.